app/views/softdrinksindustrylevy/variations/retrieve_summary.scala.html

```diff
@@ -112,84 +112,87 @@ <h1 class="heading-large">@Messages("sdil.variation.heading")</h1>
         }
     }
 
-    @if(variationData.updatedProductionSites.nonEmpty) {
-        @declaration_table("declaration", "production-sites") {
-            @variationData.updatedProductionSites.map { address =>
+    @if(variationData.producer.isProducer || variationData.copackForOthers) {
+        @if(variationData.updatedProductionSites.nonEmpty) {
+            @declaration_table("declaration", "production-sites") {
+                @variationData.updatedProductionSites.map { address =>
+                    <tr>
+                        <td>
+                            @address.nonEmptyLines.map { a =>
+                                <div>@a</div>
+                            }
+                        </td>
+                        <td class="change-answer">
+                            <a href="@sdil.controllers.variation.routes.ProductionSiteVariationController.show()">
+                            @Messages("sdil.declaration.change-link")
+                            </a>
+                        </td>
+                    </tr>
+                }
+            }
+        } else {
+            @declaration_table("declaration", "production-sites") {
                 <tr>
-                    <td>
-                        @address.nonEmptyLines.map { a =>
-                            <div>@a</div>
-                        }
-                    </td>
                     <td class="change-answer">
                         <a href="@sdil.controllers.variation.routes.ProductionSiteVariationController.show()">
-                        @Messages("sdil.declaration.change-link")
+                        @Messages("sdil.variation.add-link")
                         </a>
                     </td>
                 </tr>
             }
         }
-    }else {
-        @declaration_table("declaration", "warehouses") {
-            <tr>
-                <td class="change-answer">
-                    <a href="@sdil.controllers.variation.routes.ProductionSiteVariationController.show()">
-                    @Messages("sdil.variation.add-link")
-                    </a>
-                </td>
-            </tr>
-        }
     }
-
-    @if(variationData.updatedWarehouseSites.nonEmpty) {
-        @declaration_table("declaration", "warehouses") {
-            @variationData.updatedWarehouseSites.map { address =>
+    @if(variationData.producer.isProducer || variationData.imports || variationData.copackForOthers) {
```

Comment:
the close brace for this is in the wrong place, which means if a user is voluntary, they also don't get to see/change their contact details